src/main/java/com/mapzen/valhalla/Route.kt

```diff
@@ -278,25 +340,33 @@ public open class Route {
         }
     }
 
-    private fun snapTo(turnPoint: Node, location: Location): Location {
-        if (java.lang.Double.compare(turnPoint.lat, location.latitude) == 0
-                && java.lang.Double.compare(turnPoint.lng, location.longitude) == 0) {
-            updateDistanceTravelled(turnPoint)
-            location.bearing = turnPoint.bearing.toFloat()
+    /**
+     * asdf
+     *
+     *  @param node Current node user is at along poly line (potentially near a turn along route)
+     *  @param location Current location of user
+     *  @return Location along route to snap to
+     */
+    private fun snapTo(node: Node, location: Location): Location {
+        // if lat/lng of node and location are same, just update location's bearing to node
+        // and snap to it
+        if ((node.lat == location.latitude) && (node.lng == location.longitude)) {
```

Comment:
Realizing now that a lot of this code was generated by the built-in Java to Kotlin converter tool and has not been touched since. This is so much better!

That said I wonder if we could have a threshold value here (ex. .00001)  as well that says these two locations are basically the same and if so save us a bunch of calculations down the line. What do you think?